### PR TITLE
feat: add new tool for recovering data

### PR DESCRIPTION
**Pull request by dl239**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

### CODE REVIEW

File: tools/openmldb_ops.py [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+            continue
+        for name in tables:
+            RecoverTable(executor, db, name)
+
+    if auto_failover:
+        if not executor.SetAutofailover("true").OK():
+            log.warn("set auto_failover failed")
+
+def Rebalance(executor, new_endpoint):
+    # will not rebalance system tables
```
Comment (dl239) [inline tools/openmldb_ops.py]: migrate leader

File: tools/openmldb_ops.py [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+            return Status(-1, "get table failed")
+        for partition in result:
+            all_dict.setdefault(partition.GetEndpoint(), [])
+            all_dict[partition.GetEndpoint()].append(partition)
+            endpoint_partition_map.setdefault(partition.GetEndpoint(), set())
+            endpoint_partition_map[partition.GetEndpoint()].add(partition.GetKey())
+            db_map.setdefault(partition.GetKey(), db)
+            replica_map.setdefault(partition.GetKey, 0)
+            replica_map[partition.GetKey] += 1
+    for record in result:
```
Comment (zhanghaohit) [inline tools/openmldb_ops.py]: the `result` here is the `result` in line 281 rather than line 266

File: tools/openmldb_ops.py [hunk 3]
```diff
@@ -0,0 +1,350 @@
...
+        if not status.OK():
+            log.error(f"get all table failed")
+            return
+        for name in tables:
+            if not RecoverTable(executor, db, name).OK():
+                return
+
+def MigratePartition(db : str, partition : Partition, src_endpoint : str, desc_endpoint : str, one_replica : bool) -> Status:
+    if partition.IsLeader():
+        if one_replica and not executor.AddReplica(db, partition.GetName(), partition.GetPid(), desc_endpoint).OK():
```
Comment (zhanghaohit) [inline tools/openmldb_ops.py]: if we add replica in the `desc_endpoint`, we cannot migrate it to `desc_endpoint`, as we don't allow replicas of the same partition co-existing in the same tablet?

File: tools/openmldb_ops.py [hunk 4]
```diff
@@ -0,0 +1,378 @@
...
+    for endpoint in endpoints:
+        if not ScaleInEndpoint(executor, endpoint, alive_endpoints).OK():
+            return
+    log.info("execute scale-in success")
+
+if __name__ == "__main__":
+    (options, args) = parser.parse_args()
+    if options.cmd not in ["recoverdata", "scalein", "scaleout"]:
+        log.error(f"unsupported cmd {options.cmd}")
+        exit
```
Comment (zhanghaohit) [inline tools/openmldb_ops.py]: the `exit` seems to have no effect. maybe use `sys.exit()`

File: tools/tool.py [hunk 5]
```diff
@@ -0,0 +1,355 @@
...
+            while error_try_times > 0:
+                status, result = self.ShowOpStatus(database, name, pid)
+                error_try_times -= 1
+                if status.OK():
+                    break
+                elif error_try_times == 0:
+                    return Status(-1, "fail to execute showopstatus")
+            is_finish = True
+            for record in result:
+                if record[4] == 'kDoing' or record[4] == 'kInited':
```
Comment (zhanghaohit) [inline tools/tool.py]: if `kFailed`, should return `non-ok` status

Comment (dl239) [reply tools/tool.py:346]: there may have history failed jobs.

**Review by zhanghaohit — APPROVED** [review]

**Review by lqy222 — APPROVED** [review]

**Pull request merged**